iscsidev/iscsi.go

```diff
@@ -69,8 +70,17 @@ func Volume2ISCSIName(name string) string {
 	return strings.Replace(name, "_", ":", -1)
 }
 
-func GetTargetName(name string) string {
-	return "iqn.2019-10.io.longhorn:" + Volume2ISCSIName(name)
+func GetTargetName(volumeName string) string {
+	return "iqn.2019-10.io.longhorn:" + Volume2ISCSIName(volumeName)
+}
+
+func (dev *Device) ReloadTargetID() error {
+	tid, err := iscsi.GetTargetTid(dev.Target)
+	if err != nil {
+		return err
+	}
+	dev.targetID = tid
```

Comment:
Why target ID will be changed?

Reply:
For the upgrading engine, this target id field is not set, which is one upgrade bug. We are figuring out the target ID and setting the field instead of modifying the iscsi target.